### ISSUE TITLE
Support for runtime dynamic slices / tensor shapes

### ISSUE THREAD

**Issue by ScholliYT** [tensorflow/tflite-micro#867]
In my specific use case I would like to use slices (i.e. `StridedSlice`) on Tensors with `begin:end` being defined by values provided at runtime. Looks like this is currently not supported in TFLM 🥺

## Example
Here is a small artificial example to demonstrate the usage of a slice being parameterized by a value passed to the model by a second input. 
Example: [Colab Notebook](https://colab.research.google.com/drive/1CX5eMlzzSyPp06a4jDS0E5txaxNjmQh-?usp=sharing)  
TF-Lite Model: [Google Drive](https://drive.google.com/file/d/1sNI9cCrwLIWzgceAYiWkH-1adbjywnty/view?usp=sharing)

## Motivation
In this example the functionality has no significant use, but later I intend to make whole conv layers dynamic (in terms of number of channels) where this will be needed to implement Slimmable Conv Layers like in https://github.com/JiahuiYu/slimmable_networks. The approaches demonstrated here would allow to account for dynamic constraints on inference time without having to deploy multiple models on the device.

Feel free to ask me for clarification here if needed!

## TLDR
**In essence, I would like to do the following:**
```python
sliced_tensor = large_value[:some_value_defined_at_runtime]
```

## Current Error / Not supported functionality 
Running the Model with TFLM results in the following runtime Error:

```
tensorflow/lite/micro/kernels/strided_slice.cc:116 output_shape->data[shape_size] != dim_shape (1 != 0)
Node STRIDED_SLICE (number 7f) failed to prepare with status 1
```

which is caused by the following [line in strided_slice.cc](https://github.com/tensorflow/tflite-micro/blob/e2465c5390e288f1b22d71cb6ab2b02769c7c0e0/tensorflow/lite/micro/kernels/strided_slice.cc#L116).

Since this error already occurs at startup in prepare, I think that my attempt here is already fundamentally doing something unsupported. This error message also makes perfect sense, because at the start time the exact values of the shapes are not yet clear, since they depend on runtime values.

## Question
**Is this even supported / intended by TFLM?** Like runtime dynamic tensor shapes and so on... I can imagine that this feature would be hard to go along with the memory allocation process.
